src/utils/url/getAbsoluteUrl.ts

```diff
@@ -2,8 +2,8 @@
  * Returns an absolute URL based on the given path.
  */
 export function getAbsoluteUrl(path: string, baseUrl?: string): string {
-  // Ignore absolute URLs.
-  if (!path.startsWith('/')) {
+  // Ignore Full URL
+  if (path.startsWith('http') || path.startsWith('https')) {
```

Comment:
Is there an issue you're encountering related to this change?

So far this introduces a degression of logic: absolute URLs do not necessarily start with HTTP/HTTPS protocol. There are also `ftp://`, `data://` and [many other URL schemes](https://en.wikipedia.org/wiki/List_of_URI_schemes). Instead of accounting for each individual scheme, we reverse the predicate, checking if the URL is relative. 

I'd like to learn more about what you're trying to solve by this.

Reply:
i just want to use the path without first `/`, like `api/xxx`
for the next, i'd like to set prefix as `basename` to complete the path
```js
// handlers
import { rest } from 'msw';
export const handlers = [
  rest.get('api/xxx', (req, res, ctx) => {
       // handle request
  })
]

// entry
import { worker } from './mock/browser';
worker.start({
  resolutionContext: {
     baseUrl: `${location.origin}/`
  }
});
```
i notice there is `resolutionContext` inside handleRequest

----------

for absolute URL check.
`protocol RegExp` could be better solution

Reply:
I wonder what benefits do you gain from the setup above? The behavior you get is the default behavior in MSW: all relative requests are resolved against the current origin. A request is considered relative if its URL starts from a leading slash "/". Do you have any reasons not to write request URLs that way?

Reply:
in my case, the site location is dynamic, from `/` to `/subpath/`, to prevent change the path all over the project, i find set up a `<base href="/">`, and set api as relative.
```js
// no slash start
export const API_USER_INFO = 'api/user_info';
```
in MSW, i need change every handler path
```js
import { rest } from 'msw';
import { API_USER_INFO } from 'api/user_info';

export const handlers = [
  rest.get(`/${API_USER_INFO}`, ....)
]
```
so, i think if msw provide a default prefix would be convenient. then i found change `getAbsoluteUrl` is easier way to fulfill my request.
```diff
import { rest } from 'msw';
import { API_USER_INFO } from 'api/user_info';

export const handlers = [
-  rest.get(`/${API_USER_INFO}`, ....)
+  rest.get(API_USER_INFO, ...)
]
```

Reply:
@kettanaito should this request continue？

Reply:
@MarvelSQ, I think your use case makes sense and we should not treat `one/two` as an absolute URL. You can proceed with the fix, I've left a few comments.